pds4indextools/pds4_create_xml_index.py

```diff
@@ -222,52 +278,209 @@ def process_schema_location(file_path):
     return xsd_urls
 
 
-def process_tags(xml_results, key, root, namespaces, prefixes, args):
-    """Process XML tags based on the provided options.
+def process_headers(label_results, key, root, namespaces, prefixes):
+    """
+    Process headers to have more readable contents.
+
+    Processes XPath headers by converting parts of the XPath into element tags,
+    replacing namespaces with prefixes, and updating the label_results dictionary.
+    If a duplicate XPath is encountered, it appends an underscore and a number
+    to make the XPath unique.
+
+    Args:
+        label_results (dict): A dictionary containing XML data to be processed.
+        key (str): The key representing the XML tag to be processed.
+        root (Element): The root element of the XML tree.
+        namespaces (dict): A dictionary containing XML namespace mappings.
+        prefixes (dict): A dictionary containing XML namespace prefixes.
+    """
+    key_new = convert_header_to_xpath(root, key, namespaces)
+    for namespace in prefixes.keys():
+        if namespace in key_new:
+            key_new = key_new.replace(
+                '{'+namespace+'}', prefixes[namespace]+':')
+    if key_new in label_results:
+        # If the XPath already exists, append an underscore and a number
+        i = 1
+        while True:
+            new_key = f"{key_new}_{i}"
+            if new_key not in label_results:
+                key_new = new_key
+                break
+            i += 1
```

Comment:
This code is fine, but if you want something a little more Pythonic where you don't have to keep track of the index separately, you can use `itertools`:
```
for suffix_num in itertools.count(start=1, step=1):
    <body>
```